iothub/device/src/ConnectionStatusChangeReason.cs

```diff
@@ -8,13 +8,55 @@ namespace Microsoft.Azure.Devices.Client
     /// </summary>
     public enum ConnectionStatusChangeReason
     {
+        /// <summary>
+        /// This signifies that the client is connected, and ready to be used.
+        /// This connection status change reason is returned with a connection status of <see cref="ConnectionStatus.Connected"/>.
+        /// </summary>
         Connection_Ok,
+
+        /// <summary>
+        /// This signifies that the SAS token associated with the client has expired, and cannot be renewed.
+        /// The supplied credentials need to be fixed before a connection can be established.
+        /// NOTE: This status change reason is currently not used in the client library.
```

Comment:
Even when user supplied sas token is expired? What do we throw then?

Reply:
I would think Bad_credential, but let me run a quick test to confirm this.

Reply:
Yes, it does get mapped to bad_credential.